app/screens/LoginScreen.tsx

```diff
@@ -75,9 +78,6 @@ const LoginScreen = () => {
       });
 
     if (!direct && !addressEntered) {
-      setAddressEntered(true);
-      setIsProcessing(false);
-      passwordRef.current?.focus();
       return;
     }
```

Comment:
The whole signMessage fetching should be in this if, it's only required if address was not entered already.